service/drive.py

```diff
@@ -0,0 +1,225 @@
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+import os
+from dotenv import load_dotenv
+from google.oauth2 import service_account
+from datetime import datetime, timezone
+from audio_service import translate_with_whisper
+from summarizer import summarize_using_openai
+from logger import logger
+import ssl
+import httplib2
+from google_auth_httplib2 import AuthorizedHttp
+import time
+import logging
+
+load_dotenv()
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = [
+    "https://www.googleapis.com/auth/drive.metadata.readonly",
+    "https://www.googleapis.com/auth/drive.readonly",
+    "https://www.googleapis.com/auth/drive.file",
+    "https://www.googleapis.com/auth/drive"
+]
+
+def get_credentials():
+    """Get Google Drive API credentials."""
+    credentials = service_account.Credentials.from_service_account_file(
+        os.getenv("SERVICE_ACCOUNT_FILE"), scopes=SCOPES)
+    return credentials
+
+def user_account_creds():
+    '''Get user account credentials'''
+    creds = None
+    if os.path.exists("token.json"):
+        creds = Credentials.from_authorized_user_file("token.json", SCOPES)
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                os.getenv("GOOGLE_ACCOUNT_FILE"),
+                SCOPES
+            )
+            creds = flow.run_local_server(port=0)
+        with open("token.json", "w") as token:
+            token.write(creds.to_json())
+    return creds
+    
+def get_folder_id(service, folder_name):
+    """Get the ID of a folder by name."""
+    try:
+        results = (
+            service.files()
+            .list(q=f"mimeType='application/vnd.google-apps.folder' and name='{folder_name}'", fields="files(id, name)")
+            .execute()
+        )
+        items = results.get("files", [])
+        if not items:
+            logger.info(f"No folder found with the name '{folder_name}'.")
+            return None
+        return items[0]['id']
+    except HttpError as error:
+        logger.warning(f"An error occurred: {error}")
+        return None
+
+def get_files_in_folder(service, folder_id):
+    """Get all files in a specific folder by folder ID."""
+    try:
+        results = (
+            service.files()
+            .list(q=f"'{folder_id}' in parents", fields="nextPageToken, files(id, name, createdTime, modifiedTime, size, mimeType, webViewLink)")
+            .execute()
+        )
+        items = results.get("files", [])
+        return items
+    except HttpError as error:
+        logger.warning(f"An error occurred: {error}")
+        return []
+    
+def make_file_public(service, file_id):
```

Comment:
Do we still need this method?

Reply:
Yes, We still need it. We need to make the file publicly accessible first.